### ISSUE TITLE
[0.9.6.1] TSM1 `error writing snapshot: error opening new segment file for wal (1): invalid argument`

### ISSUE THREAD

**Issue by zstyblik** [influxdata/influxdb#5191]
Hello,

it seems like 0.9.6.1 with TMS1 just crashed on me. I guess the following isn't good state of things to be in, is it?

```
Dec 21 14:00:40 coreos01 docker[1194]: [tsm1wal] 2015/12/21 14:00:40 tsm1 WAL starting with 10485760 segment size
Dec 21 14:00:40 coreos01 docker[1194]: [tsm1wal] 2015/12/21 14:00:40 tsm1 WAL writing to /data/wal/testdb2/1m/29
[...]
Dec 21 14:25:59 coreos01 docker[1194]: [tsm1] 2015/12/21 14:25:59 error writing snapshot: error opening new segment file for wal (1): open /data/wal/telegraf/default/25/_00002.wal: no such file or directory
Dec 21 14:25:59 coreos01 docker[1194]: [http] 2015/12/21 14:25:59 10.0.3.1 - - [21/Dec/2015:14:25:59 +0000] POST /write?db=testdb2&precision=s HTTP/1.1 204 0 - python-requests/2.4.3 CPython/2.7.9 Linux/4.3.0-040300-generic c1b25af4-a7ee-11e5-bbb4-000000000000 4.809103ms
Dec 21 14:26:00 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:00 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:00 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:00 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:01 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:01 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:02 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:02 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:03 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:03 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:04 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:04 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:05 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:05 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:06 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:06 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:07 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:07 error writing snapshot: error opening new segment file for wal (1): invalid argument
Dec 21 14:26:08 coreos01 docker[1194]: [tsm1] 2015/12/21 14:26:08 error writing snapshot: error opening new segment file for wal (1): invalid argument
[... AND SO ON ...]
```

Just to be sure, `config.toml` section:

```
[data]
  dir = "/data/db"

  # Controls the storage engine used for new shards. Engines available are b1,
  # bz1, and tsm1. b1 was the original default engine from 0.9.0 to 0.9.2. bz1
  # has been the default engine since 0.9.3. tsm1 was introduced in 0.9.5 and is
  # currently EXPERIMENTAL. Consequently, data written into the tsm1 engine may
  # need to be wiped between upgrades.
  engine ="tsm1"

  # The following WAL settings are for the b1 storage engine used in 0.9.2. They won't
  # apply to any new shards created after upgrading to a version > 0.9.3.
  max-wal-size = 104857600 # Maximum size the WAL can reach before a flush. Defaults to 100MB.
  wal-flush-interval = "10m" # Maximum time data can sit in WAL before a flush.
  wal-partition-flush-delay = "2s" # The delay time between each WAL partition being flushed.

  # These are the WAL settings for the storage engine >= 0.9.3
  wal-dir = "/data/wal"
  wal-logging-enabled = true
```

Thanks.

**Comment by zstyblik** [comment]
Hm, this could be connected to CQs not being executed(https://github.com/influxdb/influxdb/issues/5190). Restart made the issue to disappear and it seems like CQs are being executed now \o/

Also, looking at one measurement(out of many, though), I see no data loss. So, may be it's a good question to ask what's this error about? :)

**Comment by jwilder** [comment]
It looks like `/data/wal/telegraf/default/25/` is not there.  Was the `default` retention policy dropped or updated?  I'm wondering if retention policy enforcement removed it, but there was still data in the cache to be snapshotted.  When the snapshot runs, it fails w/ `invalid argument` because it can't rollover to the next WAL segment.

**Comment by zstyblik** [comment]
@jwilder yes, I'm fairly sure(although, I can't verify it) default retention policy was removed at some point and may be even several times. This was a rather long-running instance of influxdb.

**Comment by jwilder** [comment]
@zstyblik OK.  Not sure if we handle that case in the code right now, should be easy to test and fix.  Restarting seems to be the best workaround for now.

**Comment by zstyblik** [comment]
@jwilder I think there were lots of DB/retention/CQ deletes and setups and deletes. Basically playing around and trying to get something going. :) But yeah, restarting influxdb has made the issue to go away.

**Comment by freeseacher** [comment]
Got that error yesterday. on 1.0.0-beta1. i have logs form container do you need them ?

**Comment by larrywalker** [comment]
I am seeing this with 0.13 as well.  
[tsm1] 2016/07/13 08:15:37 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:37 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:38 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:38 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:39 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:39 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:40 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:40 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:41 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:41 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:42 error writing snapshot: error opening new segment file for wal (1): invalid argument
[tsm1] 2016/07/13 08:15:42 error writing snapshot: error opening new segment file for wal (1): invalid argument

A restart made it go away for a day, but back again this morning.  Now a restart is failing.
[httpd] 2016/07/13 08:19:34 Starting HTTP service
[httpd] 2016/07/13 08:19:34 Authentication enabled: false
run: open server: open service: listen tcp :8086: socket: too many open files

**Comment by freeseacher** [comment]
@larrywalker maybe problem is simple ? too many open files ?

**Comment by larrywalker** [comment]
Yes, May database has grown beyond the ulimit value.  I increased the ulimit and it started.

**Comment by jwilder** [comment]
Fixed by #7024
